scheduling/tasks.go

```diff
@@ -0,0 +1,120 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type TaskId int32 // need to determine how we want to create IDs
+type RunId int32  // need to determine how we want to create IDs
+type TaskType string
+
+const (
+	ResourceCreation TaskType = "ResourceCreation"
+	HealthCheck      TaskType = "HealthCheck"
+	Monitoring       TaskType = "Monitoring"
+)
+
+type TargetType string
+
+const (
+	ProviderTarget    TargetType = "Provider"
+	NameVariantTarget TargetType = "NameVariant"
+)
+
+type Provider struct {
+	Name       string     `json:"name"`
+	TargetType TargetType `json:"targetType"`
+}
+
+type NameVariant struct {
+	Name       string     `json:"name"`
```

Comment:
what is this? is the name the entirety of the namevariant? is it a concatenation of those two? Needs clarification

Reply:
Im not sure. @sdreyer ?

Reply:
Name variant should have a name and variant

Reply:
Side note, this is just a placeholder since we'd want to use the metadata namevariant but need to figure out the circular dependecies

Reply:
nice, there are actually a few places where there are things in metadata that I've wanted to use in other packages. I started the refactor but it's quite a bit of work to change it everywhere. I think the right solution is
rename the helpers package to common (and put lib in there too)
move a lot of common shit into it, one of it being like the first 100 lines of metadata.go and maybe a few first lines of client.go

Reply:
Agreed

Reply:
For metadata doing /metadata/common and metadata/scheduling is also an option since they're treated like individual packages